src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/RuntimeInformation.Windows.cs

```diff
@@ -31,25 +31,40 @@ public static string OSDescription
             }
         }
 
-        public static Architecture OSArchitecture
+        public static unsafe Architecture OSArchitecture
         {
             get
             {
                 int osArch = s_osArchPlusOne - 1;
 
                 if (osArch < 0)
                 {
-                    Interop.Kernel32.SYSTEM_INFO sysInfo;
-                    unsafe
+                    // If we are running an x64 process on a non-x64 windows machine, we will report x64 as OS architecutre.
+                    //
+                    // IsWow64Process2 is only available on Windows 10+, so we will perform run-time introspection via indirect load
+                    if (NativeLibrary.TryGetExport(NativeLibrary.Load(Interop.Libraries.Kernel32), "IsWow64Process2", out IntPtr isWow64Process2Ptr) && isWow64Process2Ptr != IntPtr.Zero)
```

Comment:
```suggestion
                    if (NativeLibrary.TryGetExport(NativeLibrary.Load(Interop.Libraries.Kernel32), "IsWow64Process2", out IntPtr isWow64Process2Ptr))
```
The pointer is going to be non-null when TryGetExport returns true.